src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -73,7 +74,22 @@ private Database Database
                         {
                             // Reuse existing connection 
                             Server server = new Server(this.serverConnection);
-                            this.database = new Database(server, this.serverConnection.DatabaseName);
+                            // The default database name is the database name of the server connection
+                            string dbName = this.serverConnection.DatabaseName;                          
+                            if (this.connectionInfo != null)
+                            {
+                                // If there is a query DbConnection, use that connection to get the database name
+                                // This is preferred since it has the most current database name (in case of database switching)
```

Comment:
It might be better for a backlog task, but managing the DatabaseName for the SqlConnection opened in ConnectedBindingQueue might be a cleaner solution.